crates/ty_python_semantic/src/types.rs

```diff
@@ -6817,6 +6900,9 @@ impl<'db> KnownInstanceType<'db> {
                         field.default_type(self.db).display(self.db).fmt(f)?;
                         f.write_str("]")
                     }
+                    KnownInstanceType::NewType(declaration) => {
+                        f.write_str(declaration.name(self.db))
```

Comment:
I think this should be something like

```suggestion
                        write!(f, "<NewType pseudo-class '{}'>, declaration.name(self.db))
```

So that the display of a `NewType` pseudo-class clearly differentiates it from an "instance of that pseudo-class". Otherwise the output of `reveal_type` for something like this is quite confusing:

```py
from typing import NewType, reveal_type

X = NewType("X", int)
reveal_type(X)         # `X`
reveal_type(X(42))     # Also `X` currently, but the object returned by `X(42)`
                       # inhabits a very different type in our model to the object `X` itself!
```